cosmos/airflow/graph.py

```diff
@@ -561,6 +562,8 @@ def _add_producer_watcher(
         producer_task_args["selector"] = render_config.selector
         producer_task_args["exclude"] = render_config.exclude
 
+    producer_task_args["test_behavior"] = test_behavior
```

Comment:
@pankajastro It feels we changed several parts of the code interface that did not need to be changed. This PR:

- Exposed the concept of `test_behavior` to the producer task
- Modified the base operator to handle a new `kwarg`, `dbt_flag_no_test`

Did you consider simply extending the producer kwargs "exclude" list to contain  `"test_type:*`"? This could be done in `_add_producer_watcher`, which already receives `render_config`.